mediacloud/mediawords/crawler/download/feed/ap.py

```diff
@@ -0,0 +1,207 @@
+        The value of all returns all available formats (this is the default).
+
+        page_size: The maximum number of items to return per page. The default is 10 items with a maximum of 100 per page.
+
+        versions: Specifies whether to return all available versions of the content item and all ANPA filings or only the latest
+        (the same story in the ANPA format may be filed multiple times; for example, with a different category code).
+
+        REQUEST HEADERS:
+
+        Accept-Encoding: Compresses the response to the gzip format. The valid value is gzip.
+
+        """
+        url = 'https://api.ap.org/media/v/content/feed'.format(self.api_version)
+        params = {}
+        params['apikey'] = self.api_key
+        params.update(kwargs)
+        feed_data = self._make_request(url,params)
+        return json.loads(feed_data)['data']
+
+    def content(self,path,**kwargs) -> dict:
+        """Content API endpoint (Documentation: https://api.ap.org/media/v/docs/api/Content-Item/)
+        Example: https://api.ap.org/media/v[{version}]/content/{item_id}?apikey={apikey}[{optional_parameters}]
+
+        METHOD: GET
+
+        ENDPOINT PARAMETERS:
+
+        qt: Unknown. They are present in the feed response but don't appear to be in the documentation
+
+        et: Unknown. Same as above.
+
+        REQUEST HEADERS:
+
+        Accept-Encoding: Compresses the response to the gzip format. The valid value is gzip.
+        """
+
+        url = 'https://api.ap.org/media/v/content/{}'.format(path)
+
+        params = {}
+        params['apikey'] = self.api_key
+        params.update(kwargs)
+        content_data = self._make_request(url,params)
+        return content_data
+
+    def _make_request(self,
+                      url: str,
+                      params: dict = None) -> str:
+        """Internal method for making API requests"""
+
+        retries = self.retry_limit
+
+        # Begin making request and retry up to retry limit
+        while retries:
+
+            logger.debug("Making request to {} with parameters {}".format(url,params))
+
+            try:
+                response = requests.get(url,params=params)
+            except Exception as e:
+                logger.warning("Encountered an exception while making request to {}. Exception info: {}".format(url,e))
+            else:
+                if response.status_code == 200:
+                    logger.debug("Successfully retrieved {}".format(url))
+                    return response.content
+                else:
+                    logger.warning("Received HTTP status code {} when fetching {}".format(response.status_code,url))
+
+            retries -= 1
+
+            if retries == 0:
+                logger.error("Could not fetch {} after {} attempts. Giving up.".format(url,self.retry_limit))
+                raise ValueError("Could not fetch {} after {} attempts. Giving up.".format(url,self.retry_limit))
```

Comment:
ideally we would use a custom error class here such as 'McAPFetchError' or whatever